### PR TITLE
chore: fix tests for disabled RBAC

### PR DESCRIPTION
**Pull request by AlekSi**
This commit also introduces a hidden `--json` flag for `talosctl version` command
that is not supported and should be re-worked at #907.

Refs #3852.

### CODE REVIEW

File: cmd/talosctl/cmd/talos/version.go [hunk 1]
```diff
@@ -82,8 +97,12 @@ var versionCmd = &cobra.Command{
...
 
 func init() {
-	versionCmd.Flags().BoolVar(&shortVersion, "short", false, "Print the short version")
-	versionCmd.Flags().BoolVar(&clientOnly, "client", false, "Print client version only")
+	versionCmd.Flags().BoolVar(&versionCmdFlags.shortVersion, "short", false, "Print the short version")
+	versionCmd.Flags().BoolVar(&versionCmdFlags.clientOnly, "client", false, "Print client version only")
+
+	// TODO remove when https://github.com/talos-systems/talos/issues/907 is implemented
+	versionCmd.Flags().BoolVar(&versionCmdFlags.json, "json", false, "")
+	cli.Should(versionCmd.Flags().MarkHidden("json"))
```
Comment (AlekSi) [inline cmd/talosctl/cmd/talos/version.go:105]: The alternative is to call an API in the CLI integration test that sounds strange

**Review by smira — APPROVED** [review]

**Pull request merged**